python/graphstorm/gpartition/random_partition.py

```diff
@@ -45,18 +45,40 @@ class RandomPartitionAlgorithm(LocalPartitionAlgorithm):
     def _assign_partitions(self, num_partitions: int, partition_dir: str):
         num_nodes_per_type = self.metadata_dict["num_nodes_per_type"]
         ntypes = self.metadata_dict["node_type"]
-        # Note: This assumes that the order of node_type is the same as the order num_nodes_per_type
+        assert len(num_nodes_per_type) == len(ntypes), \
```

Comment:
In what case will it be non-equal? Even if there is no such node in one node type, there will be a zero here, right?